[kumar2215] iP

kumar2215 opened the pull request:
## RoverPro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

RoverPro frees your mind of having to remember things you need to do. It's,
+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/kumar2215/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 💯

And it is **FREE**!

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public final class Rover {
    public static void main(String[] args) {
        new Rover("data/Rover.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kumar2215 committed:
Modify .gitignore

kumar2215 committed:
Rename main class

kumar2215 committed:
Add greeting

kumar2215 committed:
Add support for echo

kumar2215 committed:
Add support for list

kumar2215 committed:
Add support for marking tasks

kumar2215 committed:
Tidy up code

kumar2215 committed:
Add support for different tasks

kumar2215 committed:
Change Duke to Rover

kumar2215 committed:
Add testing

kumar2215 committed:
Add error handling

kumar2215 committed:
Fix bugs

kumar2215 committed:
Add support for deleting

kumar2215 committed:
Add TaskAction enum

kumar2215 committed:
Convert static methods to be non-static

kumar2215 committed:
Add support for saving tasks

kumar2215 committed:
Modify .gitignore to not include data dir

kumar2215 committed:
Refactor getTaskString method

kumar2215 committed:
Merge branch 'branch-Level-7'

kumar2215 committed:
Add support for datetimes for tasks

kumar2215 committed:
Merge branch 'branch-Level-8'

kumar2215 committed:
Refactor: Add more OOP

kumar2215 committed:
Move classes to their respective packages

kumar2215 committed:
Refactor: create command classes for each command

kumar2215 committed:
Merge branch 'A-Packages'

kumar2215 committed:
Add option for user to retry saving

kumar2215 committed:
Modify load method to not throw exceptions

kumar2215 committed:
Merge branch 'add-gradle-support'

kumar2215 committed:
Setup Gradle and checkstyle

kumar2215 committed:
Add JUnit Testing and bug fixes

kumar2215 committed:
Update test ui testing

kumar2215 committed:
Update build.gradle

kumar2215 committed:
Add JavaDoc for most classes in src/main

kumar2215 committed:
Add xml config files

kumar2215 committed:
Add xml config files

kumar2215 committed:
Tweak code to follow coding standard

kumar2215 committed:
Add support for finding tasks by keyword

kumar2215 committed:
Merge branch 'branch-A-JavaDoc'

kumar2215 committed:
Merge branch 'branch-A-CodingStandard'

kumar2215 committed:
Merge branch 'branch-Level-9'

kumar2215 committed:
Fix issues with test inputs

kumar2215 committed:
Refactor taskList methods that show tasks

kumar2215 committed:
Update TaskListTest

kumar2215 committed:
Update FindCommand to handle empty keyword

kumar2215 committed:
Update classes to be final or sealed

darrenchooji commented on src/test/java/rover/task/EventTest.java:
```diff
@@ -0,0 +1,143 @@
+package rover.task;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
```
Any reason why you chose to 'import static' rather than 'import'?

kumar2215 replied:
It doesn't work if it's not a static import?

darrenchooji commented on src/test/java/rover/task/TaskListTest.java:
```diff
@@ -0,0 +1,209 @@
+import rover.exceptions.RoverException;
+import rover.ui.Ui;
+
+public class TaskListTest {
```
I like how comprehensive your testing is.

kumar2215 replied:
Thanks!

darrenchooji commented on src/main/java/rover/ui/Ui.java:
```diff
@@ -0,0 +1,109 @@
+    public void showHelpMessage() {
+        System.out.println(divider);
+        System.out.println("I'm sorry, but I don't know what that means.");
+        String briefHelp = """
```
I like the additional help provided, users would find this helpful.

kumar2215 replied:
Thanks!

kumar2215 committed:
Add JavaFx dependencies

kumar2215 committed:
Setup GUI while maintaining Text UI

kumar2215 committed:
Update text ui testing

kumar2215 committed:
Improve GUI

kumar2215 committed:
Make use of Varargs

kumar2215 committed:
Merge branch 'branch-Level-10'

kumar2215 committed:
Merge branch 'branch-A-Varargs'

kumar2215 committed:
Make error dialogboxes appear red

kumar2215 committed:
Fix bug with parsing RetrySaveCommand

kumar2215 committed:
Fix some issues

kumar2215 committed:
Add assertions

kumar2215 committed:
Refactor code to abide code quality standards

kumar2215 committed:
Add more usage of streams

The codebase already has some usage of streams where applicable.

More streams is being used to satisfy the A-Streams tag.

This commit adds more usage of streams in the Storage class.

Its added in the save method where the list of tasks is being written to the file
as I think that's where its most applicable.

kumar2215 committed:
Merge branch 'branch-A-Assertions'

Add assertions

kumar2215 committed:
Merge branch 'master' into branch-A-CodeQuality

kumar2215 committed:
Merge branch 'branch-A-CodeQuality'

Modify code to abide code quality standards

kumar2215 committed:
Merge branch 'master' into branch-A-Streams

kumar2215 committed:
Merge branch 'branch-A-Streams'

Add more streams

kumar2215 committed:
Update tests and code after merging

kumar2215 committed:
Set up CI

kumar2215 committed:
Update JUnit tests

kumar2215 committed:
Standardise date and time formats

kumar2215 committed:
Standardise date and time formats again

kumar2215 committed:
Modify gradle.yml

kumar2215 committed:
Modify equality of tasks to allow duplicates

kumar2215 committed:
Update tests to reflect new equality of tasks

kumar2215 committed:
Make GUI better

kumar2215 committed:
Add ability for user to personalise

kumar2215 committed:
Fix bugs

kumar2215 committed:
Add more JUnit tests

kumar2215 committed:
Add credit to code reuse

kumar2215 committed:
Update JavaDocs

kumar2215 committed:
Modify .gitignore to include jar files

kumar2215 committed:
Fix some issues

kumar2215 committed:
Update help message to include set commands

kumar2215 committed:
Add User Guide

kumar2215 committed:
Update help message to include find command

kumar2215 committed:
Update User Guide

kumar2215 committed:
Disallow assigning of past dates and times

kumar2215 committed:
Use AI to help fix some issues

kumar2215 committed:
Update JUnit tests

kumar2215 committed:
Update User Guide
